djangocms_versioning/models.py

```diff
@@ -174,6 +180,10 @@ def can_be_archived(self):
         return can_proceed(self._set_archive)
 
     def archive(self, user):
+        from .operations import send_post_version_operation, send_pre_version_operation
+
+        # trigger pre operation signal
+        action_token = send_pre_version_operation(constants.ARCHIVED, obj=self)
```

Comment:
Please decouple operation name from version state constants (use another constant, for example `constants.OPERATION_ARCHIVE` instead of `constants.ARCHIVE`)